src/foam/u2/view/ScrollDAOView.js

```diff
@@ -341,9 +245,273 @@ foam.CLASS({
     }
   ],
 
+  classes: [
+    {
+      name: 'Row',
+      extends: 'foam.u2.Element',
+
+      documentation: `Recycled DOM rows. The "view" ViewSpec is used to manually
+          tear down and rebuild row contents whenever "data" changes. Scroll
+          performance is optimized by using an inner view with no dynamic
+          bindings so that the only DOM operation is the row completely
+          replacing its contents.`,
+
+      imports: [ 'rowFormatter' ],
+
+      axioms: [
+        foam.u2.CSS.create({
+          code: function CSS() {/*
+            ^ {
+              display: block;
+              contain: layout;
+              will-change: transform;
+              padding: 5px;
+              box-sizing: border-box;
+            }
+            ^ * {
+              background-color: #dddddd;
+            }
+        */}
+        })
+      ],
+
+      properties: [
+        [ 'nodeName', 'li' ],
+        {
+          name: 'data',
+          postSet: function(old, nu) {
+            if ( this.state !== this.LOADED ) return;
+            this.el().innerHTML = this.rowFormatter.format(nu);
```

Comment:
So this is dangerous as hell.  One of the reasons we don't do anything like this in U2 was for security approval, because developers cannot be trusted to properly escape HTML every time. Just like they can't be trusted to properly manage memory.  (See endless security bugs in any C++ project, even Chrome).

I really _really_ don't think you should be side-stepping U2.  If you must, please do it in your own project and not the core framework.